Week3/homework/6-step3.js

```diff
@@ -4,3 +4,17 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 // add your solution here, or add a comment on how you would tackle this problem
+function printAllArrayItems(arr) {
+  if (typeof arr !== 'object') {
+    console.log(arr); // a break condition for recursive use
+  } else {
+    for (let i = 0; i < arr.length; i++) {
+      if (printAllArrayItems(arr[i])) {
+        console.log(printAllArrayItems(arr[i]));
+      }
+    }
+  }
+}
+
+printAllArrayItems(arr2d);
+printAllArrayItems(arr3d);
```

Comment:
Nice work. 👍 
Using recursion, very good.  So this will work no matter how deeply the array is nested.
You are the first to work from the end result (i.e. using a negative test).